module/criteriaQuestions/helper.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * name : criteriaQuestions.js
+ * author : Aman
+ * created-date : 24-Jun-2020
+ * Description : Criteria questions related information.
+ */
+
+/**
+ * CriteriaQuestionsHelper
+ * @class
+*/
+
+module.exports = class CriteriaQuestionsHelper {
+
+    /**
+   * Create Or update criteria Questions.
+   * @method
+   * @name createOrUpdate
+   * @param {String} criteriaIds - criteria ids.
+   * @param {String} [ updateQuestion = false ] - update question or criteria.
+   * @returns {JSON} - success true or false
+   */
+
+    static createOrUpdate( 
+        criteriaIds,
+        updateQuestion = false
+    ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let result = "";
+
+                if( Array.isArray(criteriaIds) ) {
+
+                    result = [];
+
+                    for (let criteria = 0; criteria < criteriaIds.length ; criteria++ ) {
```

Comment:
@aman-tunerlabs - Why do in a loop and not in a single query?

Reply:
@akash-above  Anyway loop is required for converting string of ids to ObjectId.

Reply:
Looping for the string to object is not required right, at one point you said with our without object id it will work the same way. And looping for the string is far different from looping in queries and DB interactions it is more expensive.

Reply:
@akash-above  I told for find query not for aggregate query . Please have a look at this - https://stackoverflow.com/questions/36193289/moongoose-aggregate-match-does-not-match-ids

Reply:
@akash-above  For aggregate we need to have proper objectId constructor.

Reply:
OK, but what can be achieved in one query we need not loop

Reply:
@akash-above Ok should i change it to queries.

Reply:
@akash-above  For converting string into ObjectId we need to have a loop.